pkg/monitor/agent.go

```diff
@@ -55,3 +64,77 @@ func resolveAgentType(t AgentNotification) string {
 func (n *AgentNotify) DumpInfo() {
 	fmt.Printf(">> %s: %s\n", resolveAgentType(n.Type), n.Text)
 }
+
+// PolicyUpdateNotification structures update notification
+type PolicyUpdateNotification struct {
+	Labels    []string `json:"labels,omitempty"`
+	Revision  uint64   `json:"revision,omitempty"`
+	RuleCount int      `json:"rule_count"`
+}
+
+// PolicyUpdateRepr returns string representation of monitor notification
+func PolicyUpdateRepr(rules api.Rules, revision uint64) (string, error) {
+	labels := make([]string, 0, len(rules))
+	for _, r := range rules {
+		labels = append(labels, r.Labels.GetModel()...)
+	}
+
+	notification := PolicyUpdateNotification{
+		Labels:    labels,
+		Revision:  revision,
+		RuleCount: len(rules),
+	}
+
+	repr, err := json.Marshal(notification)
+
+	return string(repr), err
+}
+
+// PolicyDeleteRepr returns string representation of monitor notification
+func PolicyDeleteRepr(deleted int, labels []string, revision uint64) (string, error) {
+	notification := PolicyUpdateNotification{
+		Labels:    labels,
+		Revision:  revision,
+		RuleCount: deleted,
+	}
+	repr, err := json.Marshal(notification)
+
+	return string(repr), err
+}
+
+// EndpointRegenNotification structures regeneration notification
+type EndpointRegenNotification struct {
+	ID     uint64   `json:"id,omitempty"`
+	Labels []string `json:"labels,omitempty"`
+	Error  string   `json:"error,omitempty"`
+}
+
+// EndpointRegenRepr returns string representation of monitor notification
+func EndpointRegenRepr(e getter.EndpointGetter, err error) (string, error) {
+	notification := EndpointRegenNotification{
+		ID:     e.GetID(),
+		Labels: e.GetLabels(),
```

Comment:
This may return an empty list of labels while in the "waiting for identity" state.
I would suggest returning `e.OpLabels.IdentityLabels()` instead.

Reply:
Done